application_sdk/server/mcp/server.py

```diff
@@ -0,0 +1,101 @@
+"""
+MCP Server implementation using FastMCP for Atlan Application SDK.
+
+This module provides the MCPServer class that automatically discovers
+activities marked with @mcp_tool decorators and mounts them on FastAPI
+using streamable HTTP transport.
+"""
+
+from typing import Any, Callable, List, Optional, Tuple, Type
+
+from fastmcp import FastMCP
+from fastmcp.server.http import StarletteWithLifespan
+
+from application_sdk.activities import ActivitiesInterface
+from application_sdk.constants import MCP_METADATA_KEY
+from application_sdk.observability.logger_adaptor import get_logger
+from application_sdk.workflows import WorkflowInterface
+
+
+class MCPServer:
+    """
+    MCP Server using FastMCP 2.0 with FastAPI mounting capability.
+
+    This server automatically discovers activities marked with @mcp_tool
+    and creates a FastMCP server that can be mounted on FastAPI.
+    """
+
+    def __init__(self, application_name: str, instructions: Optional[str] = None):
+        """
+        Initialize the MCP server.
+
+        Args:
+            application_name (str): Name of the application
+            instructions (Optional[str]): Description for the MCP server
+        """
+        self.application_name = application_name
+
+        self.logger = get_logger(__name__)
+
+        # FastMCP Server
+        self.server = FastMCP(
+            name=f"{application_name} MCP",
+            instructions=instructions,
+            on_duplicate_tools="error",
+        )
+
+    async def register_tools_from_workflow_and_activities_classes(
```

Comment:
(nit) was thinking if it'd be better if we call this `register_tools` rather? fix if you think alike, ignore otherwise.

Reply:
renamed